### ISSUE TITLE
400 error when getting RoleAssignmentScheduleInstances with FilterByCurrentUserWithOn("principal")

### ISSUE THREAD

**Issue by Shepard246** [microsoftgraph/msgraph-beta-sdk-dotnet#464]
Previously I would query active PIM roles based on current user's PrincipalId: 

      var actRoles = await graphClient.RoleManagement.DirectoryObject.RoleAssignmentScheduleInstances 
                .FilterByCurrentUserWithOn("principal")
                .GetAsync();

Within the last month I updated from 5.1.0 to 5.4.0 I now receive the following:

```
[09:11:58 ERR] The server returned an unexpected status code and no error factory is registered for this code: 400
   at Microsoft.Kiota.Http.HttpClientLibrary.HttpClientRequestAdapter.ThrowFailedResponse(HttpResponseMessage response, Dictionary`2 errorMapping)
   at Microsoft.Kiota.Http.HttpClientLibrary.HttpClientRequestAdapter.SendAsync[ModelType](RequestInformation requestInfo, ParsableFactory`1 factory, IResponseHandler responseHandler, Dictionary`2 errorMapping, CancellationToken cancellationToken)
   at Microsoft.Kiota.Http.HttpClientLibrary.HttpClientRequestAdapter.SendAsync[ModelType](RequestInformation requestInfo, ParsableFactory`1 factory, IResponseHandler responseHandler, Dictionary`2 errorMapping, CancellationToken cancellationToken)
   at EAT.Core.EOManagement.GetPIMRoles() in C:\Users\Travis.Wiese\source\repos\EAT\Core\EOManagement.cs:line 295
   at EAT.MVVM.ViewModel.PIM_RolesViewModel.<.ctor>b__56_0(Object o) in C:\Users\Travis.Wiese\source\repos\EAT\MVVM\ViewModel\PIM_RolesViewModel.cs:line 164
   at System.Threading.Tasks.Task.<>c.<ThrowAsync>b__128_0(Object state)
   at System.Windows.Threading.ExceptionWrapper.InternalRealCall(Delegate callback, Object args, Int32 numArgs)
   at System.Windows.Threading.ExceptionWrapper.TryCatchWhen(Object source, Delegate callback, Object args, Int32 numArgs, Delegate catchHandler)
   at System.Windows.Threading.DispatcherOperation.InvokeImpl()
   at MS.Internal.CulturePreservingExecutionContext.CallbackWrapper(Object obj)
   at System.Threading.ExecutionContext.RunInternal(ExecutionContext executionContext, ContextCallback callback, Object state)
--- End of stack trace from previous location ---
   at MS.Internal.CulturePreservingExecutionContext.Run(CulturePreservingExecutionContext executionContext, ContextCallback callback, Object state)
   at System.Windows.Threading.DispatcherOperation.Invoke()
   at System.Windows.Threading.Dispatcher.ProcessQueue()
   at System.Windows.Threading.Dispatcher.WndProcHook(IntPtr hwnd, Int32 msg, IntPtr wParam, IntPtr lParam, Boolean& handled)
   at MS.Win32.HwndWrapper.WndProc(IntPtr hwnd, Int32 msg, IntPtr wParam, IntPtr lParam, Boolean& handled)
   at MS.Win32.HwndSubclass.DispatcherCallbackOperation(Object o)
   at System.Windows.Threading.ExceptionWrapper.InternalRealCall(Delegate callback, Object args, Int32 numArgs)
   at System.Windows.Threading.ExceptionWrapper.TryCatchWhen(Object source, Delegate callback, Object args, Int32 numArgs, Delegate catchHandler)
   at System.Windows.Threading.Dispatcher.LegacyInvokeImpl(DispatcherPriority priority, TimeSpan timeout, Delegate method, Object args, Int32 numArgs)
   at MS.Win32.HwndSubclass.SubclassWndProc(IntPtr hwnd, Int32 msg, IntPtr wParam, IntPtr lParam)
   at MS.Win32.UnsafeNativeMethods.DispatchMessage(MSG& msg)
   at System.Windows.Threading.Dispatcher.PushFrameImpl(DispatcherFrame frame)
   at System.Windows.Threading.Dispatcher.PushFrame(DispatcherFrame frame)
   at System.Windows.Threading.Dispatcher.Run()
   at System.Windows.Application.RunDispatcher(Object ignore)
   at System.Windows.Application.RunInternal(Window window)
   at System.Windows.Application.Run()
   at EAT.App.Main()
An unhandled exception of type 'Microsoft.Kiota.Abstractions.ApiException' occurred in System.Private.CoreLib.dll
The server returned an unexpected status code and no error factory is registered for this code: 400
```

If I remove the filter, it works as expected with all results returned.

I have also tried:
```
`var reqInfo = graphClient.RoleManagement.DirectoryObject.RoleAssignmentScheduleInstances.FilterByCurrentUserWithOn("principal").CreateGetRequestInformation();`
            
var col = await graphClient.RequestAdapter.SendAsync<RoleAssignmentCollectionResponse>(reqInfo, RoleAssignmentCollectionResponse.CreateFromDiscriminatorValue);
```

Which I saw in another issue, and this produces the same error with the filter, but successful results without.

Let me know if there are further logs I can get or things I can try to assist.

**Comment by Shepard246** [comment]
Rolling back to 5.1.0-preview and removing Microsoft.Kiota.Abstractions (1.0.0-preview.8) appears to return functionality.

**Comment by andrueastman** [comment]
Thanks for raising this @Shepard246,

I believe this is due to https://github.com/microsoft/kiota/issues/1586 as the generator would incorrectly generate some of the request builders. 

With the generator fix already made, you should be able to validate this in the next version 5.5.0 coming out this week with the latest metadata.

**Comment by Shepard246** [comment]
> Thanks for raising this @Shepard246,
> 
> I believe this is due to [microsoft/kiota#1586](https://github.com/microsoft/kiota/issues/1586) as the generator would incorrectly generate some of the request builders.
> 
> With the generator fix already made, you should be able to validate this in the next version 5.5.0 coming out this week with the latest metadata.

Awesome, I'll report back once it's out. Thanks for the quick response @andrueastman!

**Comment by andrueastman** [comment]
Hey @Shepard246,

We've now released version [5.5.0-preview](https://www.nuget.org/packages/Microsoft.Graph.Beta/5.5.0-preview) with the fixes from https://github.com/microsoft/kiota/issues/1586.

I'll close this for now. Feel free to re-open in the event anything is amiss.

**Comment by Shepard246** [comment]
Awesome, confirmed fixed with this update, thank you again!
